base/compiler/abstractinterpretation.jl

```diff
@@ -222,17 +223,12 @@ function abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
         if mi !== nothing && !const_prop_methodinstance_heuristic(interp, match, mi::MethodInstance, arginfo, sv)
             csig = get_compileable_sig(method, sig, match.sparams)
             if csig !== nothing && csig !== sig
-                # The result of this inference is not directly used, so temporarily empty
```

Comment:
The code needs to check this very far from this call site (via InferenceState, not a local argument), so I don't think this is quite right still

Reply:
I don't quite understand what you're saying. In what circumstance would we be looking this up in this frame's InferenceState without having the StmtInfo created here passed down? I didn't think I saw any such case.

Reply:
I thought we checked it during recursion handling?